servant-server/src/Servant/Server/Internal.hs

```diff
@@ -351,12 +361,23 @@ instance (KnownSymbol sym, FromHttpApiData a, HasServer api context)
         -- named "foo" or "foo[]" and call parseQueryParam on the
         -- corresponding values
         parameters r = filter looksLikeParam (querytext r)
-        values r = mapMaybe (convert . snd) (parameters r)
-    in  route (Proxy :: Proxy api) context (passToServer subserver values)
+        parseParam (paramName, paramTxt) =
+          case parseQueryParam (fromMaybe "" paramTxt) of
```

Comment:
Why are we using `fromMaybe ""` here? Shouldn't we ignore empty values (like it was before with `mapMaybe`)?